includes/class-wc-google-analytics-js.php

```diff
@@ -288,13 +288,23 @@ public static function load_analytics_universal( $logged_in ) {
 	function add_transaction( $order ) {
 		if ( 'yes' == self::get( 'ga_use_universal_analytics' ) ) {
 			if ( 'yes' === self::get( 'ga_enhanced_ecommerce_tracking_enabled' ) ) {
-				return self::add_transaction_enhanced( $order );
+				$transaction_code = self::add_transaction_enhanced( $order );
 			} else {
-				return self::add_transaction_universal( $order );
+				$transaction_code = self::add_transaction_universal( $order );
 			}
 		} else {
-			return self::add_transaction_classic( $order );
+			$transaction_code = self::add_transaction_classic( $order );
 		}
+
+		// Wrap the transaction in a check against localStorage to avoid duplicate firing if page is reloaded without hitting server
+		$code = "var gaOrders = localStorage.getItem('ga-orders');";
+		$code .= "gaOrders = (gaOrders !== null) ? gaOrders : '';";
+		$code .= "if (gaOrders.split(';').indexOf('" . esc_js( $order->get_order_number() ) . "') === -1) {";
+		$code .= $transaction_code;
+		$code .= "localStorage.setItem('ga-orders', gaOrders + '" . esc_js( $order->get_order_number() ) . "');";
+		$code .= "}";
```

Comment:
May I suggest a readability upgrade here

```php
$code = sprintf( <<<JS
  var gaOrders = localStorage.getItem('ga-orders');
  gaOrders = (gaOrders !== null) ? gaOrders : '';
  if (gaOrders.split(';').indexOf('%1$s') === -1) {
    %2$s;

    localStorage.setItem('ga-orders', gaOrders + '%1$s');
  }
JS
, esc_js( $order->get_order_number() ), $transaction_code );
```

Reply:
The way it's been done is more consistent with the existing code base. I do agree it could be improved though. I'd suggest making that sort of refactor in a separate pull request to change it across the whole plugin at the same time.